rollup.config.js

```diff
@@ -47,37 +86,30 @@ const plugins = [
   commonjs({
     ignoreGlobal: true,
   }),
-  !esbundle && replace({
-    'process.env.NODE_ENV': JSON.stringify(prod ? 'production' : 'development'),
-  }),
-  prod && inject({
-    process: processShim,
-  }),
-  babel({
-    babelrc: false,
-    presets: [
-      ['env', { modules: false, loose: true }],
-      'react',
-    ],
-    plugins: [
-      !prod && 'flow-react-proptypes',
-      prod && 'transform-react-remove-prop-types',
-      'transform-flow-strip-types',
-      'external-helpers',
-      'transform-object-rest-spread',
-      'transform-class-properties',
-    ].filter(Boolean),
-  }),
+  UMD &&
+    replace({
+      'process.env.NODE_ENV': JSON.stringify(
+        PRODUCTION ? 'production' : 'development',
+      ),
+    }),
+  PRODUCTION &&
+    inject({
+      process: processShim,
+    }),
+  babel(),
 ].filter(Boolean)
 
-if (prod) plugins.push(uglify(), visualizer({ filename: './bundle-stats.html' }))
+if (PRODUCTION) {
+  plugins.push(uglify(), visualizer({ filename: './bundle-stats.html' }))
+}
 
 export default {
-  entry: 'src/index.js',
-  moduleName: 'styled',
-  external: ['react'].concat(esbundle ? Object.keys(pkg.dependencies) : []),
-  exports: 'named',
-  targets,
+  input,
+  external: ['react', 'react-native', 'react-primitives'].concat(
+    !UMD ? Object.keys(pkg.dependencies) : [],
+  ),
+  output,
   plugins,
   globals: { react: 'React' },
+  outro: BABEL_ENV === 'cjs' ? "module.exports = exports['default']" : '',
```

Comment:
Isn't this technically a breaking change? Did people using the CJS bundles have to do `const styled = require('styled-components').default` before but not now?

Reply:
Both flavours are supported now:
- `const styled = require('styled-components').default`
- `const styled = require('styled-components')`

Adding the support for latter is not a breaking change, but rather an 'enhancement'.

But anyway `babel-plugin-add-module-exports` was already used on master on cjs target - see [here](https://github.com/styled-components/styled-components/blob/470be50c1ac0138c1ff60a27aa1021db6c5ca56b/.babelrc#L8). So it would be a breaking change to remove that.

It is somewhat a breaking change to remove `lib` directory. People could import things from 'internal' `lib` directory before this PR, probably not many did this, but that's a question for you if you consider removing this a breaking change. IMHO it shouldnt be treated as such, lib structure is in my eyes an implementation detail and reaching to it should be considered a bad practice.

Reply:
Ohh I see, that makes sense! Thanks for the explanation.

Yeah definitely a bad practice, I would definitely not call that a breaking change.